source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
+
+            vertices[k] = renderer::VertexPosCol(position, color);
+        }
+
+        for (s32 i = 0; i < cylinder.vertices().size(); ++i, ++k)
+        {
+            glm::vec3 position = cylinder.vertices()[i].position;
+            glm::vec4 color = { 0.274509817f, 0.509803951f, 0.705882370f, 1.000000000f }; // Steel Blue
+
+            vertices[k] = renderer::VertexPosCol(position, color);
+        }
+
+        rsl::vector<u16> indices;
+        indices.insert(indices.end(), rsl::begin(box.indices()), rsl::end(box.indices()));
+        indices.insert(indices.end(), rsl::begin(grid.indices()), rsl::end(grid.indices()));
+        indices.insert(indices.end(), rsl::begin(sphere.indices()), rsl::end(sphere.indices()));
+        indices.insert(indices.end(), rsl::begin(cylinder.indices()), rsl::end(cylinder.indices()));
+
+        const u32 vb_byte_size = total_vertex_count * sizeof(renderer::VertexPosCol);
+        const u32 ib_byte_size = total_index_count * sizeof(u16);
+
+        auto geometry = rsl::make_unique<renderer::Mesh>();
+
+        geometry->name = "scene_geometry"_med;
+
+        renderer::commands::CreateBufferCommandDesc v_create_buffer_command_desc = create_buffer_parameters<renderer::VertexPosCol>(vertices.data(), vertices.size());
+        geometry->vertex_buffer = renderer::create_vertex_buffer(rsl::move(v_create_buffer_command_desc));
+        renderer::commands::CreateBufferCommandDesc i_create_buffer_command_desc = create_buffer_parameters<u16>(indices.data(), indices.size());
+        geometry->index_buffer = renderer::create_index_buffer(rsl::move(i_create_buffer_command_desc));
+
+        geometry->vertex_byte_stride = sizeof(renderer::VertexPosCol);
+        geometry->vertex_buffer_byte_size = vb_byte_size;
+        geometry->index_format = renderer::IndexBufferFormat::R16_UINT;
+        geometry->index_buffer_byte_size = ib_byte_size;
+
+        geometry->draw_args["box"_small] = box_submesh;
+        geometry->draw_args["grid"_small] = grid_submesh;
+        geometry->draw_args["sphere"_small] = sphere_submesh;
+        geometry->draw_args["cylinder"_small] = cylinder_submesh;
+
+        g_regina_ctx.meshes[geometry->name] = rsl::move(geometry);
+
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    bool build_cube_render_items()
+    {
+        auto cube_r_item = renderer::RenderItem();
 
-  ApplicationCreationParams app_entry(PlatformCreationParams&& platformParams)
-  {
-    return create_regina_app_creation_params(rsl::move(platformParams));
-  }
+        glm::mat4 scale = glm::scale(cube_r_item.world, glm::vec3(2.0f, 2.0f, 2.0f));
+
+        cube_r_item.world = scale;
+        cube_r_item.constant_buffer_index = 0;
+        cube_r_item.geometry = g_regina_ctx.mesh_cube.get();
+        cube_r_item.topology = renderer::PrimitiveTopology::TRIANGLELIST;
```

Comment:
We will always be drawing TriangleLists AFAIK, unless we do some debug drawing ( this is should use different API than the usual renderer we have IMO ). So eventually this will move to somewhere more abstract but as of now we are trying to draw things on the screen. We should have full control on what we are going to draw.